File: HackRank-DataStructure/Array and Linked List/Delete duplicate-value nodes from a sorted linked list.cpp
SinglyLinkedListNode* removeDuplicates(SinglyLinkedListNode* llist) {
SinglyLinkedListNode* head = llist;
while(llist != 0 && llist->next!= 0)
{
    while( llist->next != 0 && llist->data == llist->next->data )
        llist->next = llist->next->next;
    llist = llist->next;
}
return head;
}

File: HackRank-DataStructure/Array and Linked List/Delete duplicate-value nodes from a sorted linked list.py
def removeDuplicates(llist):
    head = llist
    while(llist != None and llist.next != None):
        while(llist.next != None and llist.data == llist.next.data):
            llist.next = llist.next.next
        llist = llist.next
    return head